editoast/src/views/work_schedules.rs

```diff
@@ -0,0 +1,133 @@
+use actix_web::post;
+use actix_web::web::Data;
+use actix_web::web::Json;
+use chrono::NaiveDateTime;
+use derivative::Derivative;
+use diesel_async::scoped_futures::ScopedFutureExt;
+use diesel_async::AsyncConnection;
+use editoast_derive::EditoastError;
+use serde::Deserialize;
+use serde::Serialize;
+use thiserror::Error;
+use utoipa::ToSchema;
+
+use crate::error::InternalError;
+use crate::error::Result;
+use crate::modelsv2::work_schedules::WorkSchedule;
+use crate::modelsv2::work_schedules::WorkScheduleType;
+use crate::modelsv2::Changeset;
+use crate::modelsv2::Create;
+use crate::modelsv2::Model;
+use crate::schema::TrackRange;
+use crate::DbPool;
+
+crate::routes! {
+    "/work_schedules" => {
+        create,
+    }
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "work_schedule")]
+pub enum WorkScheduleError {
+    // The work_schedule start and end date are in the wrong order
+    #[error("The work_schedule start date must be before the end date")]
+    #[editoast_error(status = 400)]
+    StartDateAfterEndDate,
+}
+
+/// This structure is used by the post endpoint to create a work schedule
+#[derive(Serialize, Deserialize, Derivative, ToSchema)]
+#[derivative(Default)]
+struct WorkScheduleCreateForm {
+    pub start_date_time: NaiveDateTime,
+    pub end_date_time: NaiveDateTime,
+    pub track_ranges: Vec<TrackRange>,
+    pub obj_id: String,
+    pub work_schedule_type: WorkScheduleType,
+}
+
+impl WorkScheduleCreateForm {
+    pub fn into_work_schedule_changeset(self) -> Result<Changeset<WorkSchedule>> {
+        let work_schedule_changeset = WorkSchedule::changeset()
+            .start_date_time(self.start_date_time)
+            .end_date_time(self.end_date_time)
+            .track_ranges(self.track_ranges)
+            .obj_id(self.obj_id)
+            .work_schedule_type(self.work_schedule_type);
+        WorkSchedule::validate(&work_schedule_changeset)?;
+        Ok(work_schedule_changeset)
+    }
+}
+
+#[utoipa::path(
+    tag = "work_schedules",
+    request_body = WorkScheduleCreateForm,
+    responses(
+        (status = 201, body = WorkSchedule, description = "The created work schedule"),
+    )
+)]
+#[post("")]
+async fn create(
+    db_pool: Data<DbPool>,
+    data: Json<WorkScheduleCreateForm>,
+) -> Result<Json<WorkSchedule>> {
+    let conn = &mut db_pool.get().await?;
+
+    let work_schedule = conn
+        .transaction::<_, InternalError, _>(|conn| {
```

Comment:
I'm not sure the transaction is needed here.

Reply:
I removed it, thanks :pray: